cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/DefaultKafkaQuotasManager.java

```diff
@@ -0,0 +1,232 @@
+            quotasPluginKafka.getControllerMutationRate() != null ? quotasPluginKafka.getControllerMutationRate() : null));
+
+        return ops;
+    }
+
+    /**
+     * Based on configuration in {@param quotasPlugin}, it configures the default user quota in Kafka.
+     *
+     * @param reconciliation            Reconciliation marker
+     * @param vertx                     Vert.x instance
+     * @param adminClientProvider       Kafka Admin client provider
+     * @param pemTrustSet               Trust set for TLS authentication in PEM format
+     * @param pemAuthIdentity           Identity for TLS client authentication in PEM format
+     * @param quotasPlugin              Configuration of Kafka quotas plugin
+     *
+     * @return  Future that completes when the default user quota configuration is completed
+     */
+    public static Future<Void> reconcileDefaultUserQuotas(
+        Reconciliation reconciliation,
+        Vertx vertx,
+        AdminClientProvider adminClientProvider,
+        PemTrustSet pemTrustSet,
+        PemAuthIdentity pemAuthIdentity,
+        QuotasPlugin quotasPlugin
+    ) {
+        LOGGER.debugCr(reconciliation, "Reconciling default user quotas in Kafka");
+        String bootstrapHostname = KafkaResources.bootstrapServiceName(reconciliation.name()) + "." + reconciliation.namespace() + ".svc:" + KafkaCluster.REPLICATION_PORT;
+
+        LOGGER.debugCr(reconciliation, "Creating AdminClient for setting default quota using {}", bootstrapHostname);
+        Admin kafkaAdmin = adminClientProvider.createAdminClient(bootstrapHostname, pemTrustSet, pemAuthIdentity);
+
+        boolean isNotKafkaPlugin = !(quotasPlugin instanceof QuotasPluginKafka);
+        QuotasPluginKafka quotasPluginKafka = isNotKafkaPlugin ? emptyQuotasPluginKafka() : (QuotasPluginKafka) quotasPlugin;
+
+        List<ClientQuotaAlteration.Op> ops = prepareQuotaConfigurationRequest(quotasPluginKafka);
+
+        return shouldAlterDefaultQuotasConfig(reconciliation, vertx, kafkaAdmin, ops, isNotKafkaPlugin)
+            .compose(shouldUpdateQuotas -> {
+                if (shouldUpdateQuotas) {
+                    Promise<Void> promise = Promise.promise();
+
+                    ClientQuotaAlteration clientQuotaAlteration = new ClientQuotaAlteration(DEFAULT_USER_ENTITY, ops);
+
+                    LOGGER.debugCr(reconciliation, "Default user quotas differ and will be updated");
+                    alterQuotas(reconciliation, vertx, kafkaAdmin, clientQuotaAlteration)
+                        .onComplete(result -> {
+                            if (result.succeeded()) {
+                                LOGGER.debugCr(reconciliation, "Successfully altered default user quotas");
+                                promise.complete();
+                            } else {
+                                LOGGER.warnCr(reconciliation, "Failed to alter default user quotas", result.cause());
```

Comment:
shouldn't it be ERROR level logging?

Reply:
Which doesn't match with a WARN level here. ERROR would be more appropriate.

Reply:
Why does it not match ... I think we decided to do it this way a long time ago. 

Anyway, I don't care much - but I told Lukas to use WANR here, so he will need to change it back if you want 🤷.

Reply:
I can change it back, if having it in error is better here.